test_workspace/recordset.bosatsu

```diff
@@ -10,9 +10,12 @@ struct RecordSet[shape](fields: shape[Field[String], Field[Int], Field[Bool]], r
 def restructure(RecordSet(fields, rows), f: forall string,int,bool. shape1[string, int, bool] -> shape2[string, int, bool]) -> RecordSet[shape2]:
   RecordSet(f(fields), rows.map_List(f))
 
-def add_int_field(RecordSet(fields, rows): RecordSet[shape1], name: String, f: shape1[String, Int, Bool] -> Int, g: forall string,int,bool. shape1[string,int,bool] -> int -> shape2[string,int,bool]):
-  RecordSet(g(fields, Field(name, RVInt)), rows.map_List(\row -> g(row, f(row))))
+struct PairIntShape[shape,string,int,bool](int: int, rest: shape[string, int, bool])
```

Comment:
I'm trying to figure out if I can make this `PairShape` instead of having a `PairIntShape,PairStringShape,PairBoolShape`... but I don't think I can